Dockerfile

```diff
@@ -1,8 +1,48 @@
 # syntax-docker/dockerfile:1
-FROM python:alpine3.14
+
+FROM python:3.11-slim AS builder
+
+# This is neccesary step for psycopg2 to function in slim image
+RUN apt-get update && \
+    apt-get install -y libpq-dev gcc
+
+# Create virtual environment
+RUN python -m venv /opt/venv
+# Activate venv
+ENV PATH="/opt/venv/bin:$PATH"
+
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+
+
+# Operational stage
+FROM python:3.11-slim
+
+RUN apt-get update && \
+    apt-get install -y libpq-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+# Get venv from bulider stage
+COPY --from=builder /opt/venv /opt/venv
+
+ENV PYTHONDONTWRITEBYTECODE=1\
+    PYTHONUNBUFFERED=1\
+    PATH="/opt/venv/bin:$PATH"
+
 WORKDIR /app
 
-COPY . .
-RUN pip3 install -r requirements.txt
+COPY . /app/
+
+# ---------------------------
+# This was previously the working dockerfile:
+# FROM python:alpine3.14
+# ENV PYTHONUNBUFFERED 1
+
+# WORKDIR /app
+
+# COPY . .
+# RUN pip3 install -r requirements.txt
+# --------------------------------------
 
-CMD [ "python3", "manage.py", "runserver", "0.0.0.0:8000" ]
\ No newline at end of file
+# This CMD is being run in the docker-compose file
+# CMD [ "python3", "manage.py", "runserver", "0.0.0.0:8000" ]
```

Comment:
Leave the CMD in.  Just because you override it in docker-compose doesn't mean you don't want it here.  With it commented out here, then `docker run` AND `kubernetes` will both just instantly exit when it starts up.